api/image_pipeline/pipeline_image_util.py

```diff
@@ -0,0 +1,660 @@
+
+    return delivery_object
+
+
+# Delivery stuff
+def delivery_repo_is_available(name: str) -> bool:
+    """
+    Function to check if the given delivery repo exists
+
+    :name: Name of the delivery repo
+    """
+    try:
+        _ = get_delivery_repo_id(name)
+        return True
+    except exceptions.DeliveryRepoIDNotFound:
+        return False
+
+
+def brew_from_delivery(delivery_repo: str) -> str:
+    """
+    Function to get the brew name from the delivery repo
+    """
+    url = f"https://pyxis.engineering.redhat.com/v1/repositories/registry/registry.access.redhat.com/repository/{delivery_repo}/images"
+    response = request_with_kerberos(url)
+
+    if response.status_code == 404:
+        raise exceptions.BrewFromDeliveryNotFound(
+            f"Brew package could not be found from delivery repo `{delivery_repo}`")
+
+    result = []
+    for data in response.json()['data']:
+        result.append(data['brew']['package'])
+
+    result = list(set(result))
+    if len(result) > 1:
+        raise exceptions.MultipleBrewFromDelivery(f"Multiple brew packages found for delivery repo `{delivery_repo}`")
+
+    return result.pop()
+
+
+def brew_to_cdn_delivery(brew_name: str, variant: str, delivery_repo_name: str) -> str:
+    """
+    Function the get the CDN name from brew name, variant and delivery repo name
+
+    :brew_name: Brew repo name
+    :variant: 8Base-RHOSE variant
+    :delivery_repo_name: Delivery repo name
+    """
+    cdn_repo_names = brew_to_cdn(brew_name, variant)
+    for cdn_repo_name in cdn_repo_names:
+        delivery = cdn_to_delivery(cdn_repo_name)
+        if delivery == delivery_repo_name:
+            return cdn_repo_name
+    raise exceptions.BrewToCdnWithDeliveryNotFound(
+        f"Could not find CDN from Brew name from delivery repo `{delivery_repo_name}`")
+
+
+def get_delivery_repo_id(name: str) -> str:
+    """
+    Function to get the delivery repo id. Used to construct the delivery repo URL to direct to its page in Pyxis.
+
+    :name: Delivery repo name
+    """
+    url = f"https://pyxis.engineering.redhat.com/v1/repositories?filter=repository=={name}"
+    response = request_with_kerberos(url)
+
+    if response.status_code == 404:
+        raise exceptions.DeliveryRepoUrlNotFound(f"Couldn't find delivery repo link on Pyxis")
+
+    try:
+        repo_id = response.json()['data'][0]['_id']
+    except Exception:
+        raise exceptions.DeliveryRepoIDNotFound(f"Couldn't find delivery repo ID on Pyxis for {name}")
+
+    return repo_id
+
+
+# Methods
+@util.refresh_krb_auth
+def request_with_kerberos(url: str) -> requests.Response():
+    # Kerberos authentication
+    kerberos_auth = HTTPKerberosAuth(mutual_authentication=OPTIONAL)
+
+    # Sending the kerberos ticket along with the request
+    response = requests.get(url, auth=kerberos_auth)
+
+    if response.status_code == 401:
+        raise exceptions.KerberosAuthenticationError("Kerberos Authentication failed")
```

Comment:
In this case, could it log the response so that we might have a chance to investigate auth errors?